rust/yarp-sys/build.rs

```diff
@@ -0,0 +1,168 @@
+use std::{
+    io::{self, Write},
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+fn main() -> Result<(), io::Error> {
+    rake_compile()?;
+
+    let ruby_build_path = ruby_build_path();
+    let ruby_include_path = ruby_include_path();
+
+    // Tell cargo/rustc that we want to link against `librubyparser.a`.
+    println!("cargo:rustc-link-lib=static=rubyparser");
+
+    // Add `[root]/build/` to the search paths, so it can find `librubyparser.a`.
+    println!(
+        "cargo:rustc-link-search=native={}",
+        ruby_build_path.display()
+    );
+
+    // This is where the magic happens.
+    let bindings = generate_bindings(&ruby_include_path);
+
+    // Write the bindings to file.
+    write_bindngs(bindings);
+
+    Ok(())
+}
+
+/// Convenience step for running `rake compile` before doing our stuff.
+///
+fn rake_compile() -> Result<(), io::Error> {
+    let output = Command::new("rake").arg("compile").output()?;
+
+    io::stdout().write_all(&output.stdout)?;
+    io::stderr().write_all(&output.stderr)?;
```

Comment:
this code can be simplified by invoking the command with `status` instead of `output`. The docs specify that the command will inherit stdout and stderr by default if invoked with either `status` or `spawn`.

- https://doc.rust-lang.org/std/process/struct.Command.html#method.new
- https://doc.rust-lang.org/std/process/struct.Command.html#method.status

Reply:
Nice.